lib/auth/auth.go

```diff
@@ -1170,6 +1170,88 @@ func (a *Server) GetMFADevices(ctx context.Context, req *proto.GetMFADevicesRequ
 	}, nil
 }
 
+// verifyMFARespAndAddDeviceWithToken validates MFA register response and on success,
+// adds the new MFA device. This method is specifically for secrets and storage keys associated with a user token.
+func (a *Server) verifyMFARespAndAddDeviceWithToken(ctx context.Context, regResp *proto.MFARegisterResponse, token types.UserToken, deviceName string) error {
+	cap, err := a.GetAuthPreference(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
```

Comment:
Check SecondFactorOff?

```suggestion
	if cap.GetSecondFactor() == constants.SecondFactorOff {
		return trace.BadParameter("second factor disabled by cluster configuration")
	}

```